File: maintenance/check_words.js
const fs = require('fs');

try {
    let code = fs.readFileSync('js/exercises.js', 'utf8');
    code += '\nmodule.exports = exercises;';
    fs.writeFileSync('temp_exercises.js', code);
    
    const exercises = require('./temp_exercises.js');
    if (!exercises.generi) {
        console.log("No generi found.");
        process.exit(0);
    }
    
    let count = 0;
    for (const [genreName, levels] of Object.entries(exercises.generi)) {
        for (const [levelName, levelArr] of Object.entries(levels)) {
            for (const item of levelArr) {
                const words = item.text.replace(/<[^>]+>/g, '').split(/[\s\n\r]+/).filter(w => w.length > 0).length;
                if (words < 600) {
                    console.log(`SHORT: ${genreName} - ${levelName} - id ${item.id}: ${words} parole`);
                    count++;
                } else {
                    console.log(`OK: ${genreName} - ${levelName} - id ${item.id}: ${words} parole`);
                }
            }
        }
    }
    console.log(`\nTotal short texts: ${count}`);
} catch (e) {
    console.error(e);
}

File: maintenance/check_words.py
import re

with open('js/exercises.js', 'r', encoding='utf-8') as f:
    content = f.read()

# Find the generi section
start_idx = content.find('"generi":')
if start_idx == -1:
    print("Could not find generi section.")
    exit(1)

# we just care about "text": "..." inside the file after "generi"
generi_content = content[start_idx:]

# Find all occurrences of "text": "..."
# Note: some texts might have escaped quotes, but our texts mostly use simple quotes or don't escape "
matches = re.finditer(r'"id": (\d+),.*?"title": "(.*?)",.*?"text": "(.*?)"', generi_content, re.DOTALL)

count = 0
for match in matches:
    id_val = match.group(1)
    title = match.group(2)
    text = match.group(3)
    
    # Remove HTML tags for counting
    clean_text = re.sub(r'<[^>]+>', '', text)
    words = clean_text.split()
    word_count = len(words)
    
    if word_count < 600:
        print(f"SHORT: id {id_val} - {title} - {word_count} parole")
        count += 1
    else:
        print(f"OK: id {id_val} - {title} - {word_count} parole")

print(f"\nTotal short texts: {count}")
